client/actions/auth.js

```diff
@@ -1,26 +1,33 @@
+import url from 'url';
 import axios from 'axios';
 import jwtDecode from 'jwt-decode';
 
+import { show as showLock } from '../utils/lock';
 import * as constants from '../constants';
 
-const auth0 = new Auth0({ // eslint-disable-line no-undef
+const webAuth = new auth0.WebAuth({ // eslint-disable-line no-undef
   domain: window.config.AUTH0_DOMAIN,
   clientID: window.config.AUTH0_CLIENT_ID,
+  responseType: 'id_token token',
+  scope: 'openid name email nickname groups roles app_metadata authorization read:applications',
+  audience: 'urn:sso-dashboard-api',
   callbackURL: `${window.config.BASE_URL}/login`,
   callbackOnLocationHash: true
 });
 
 export function login(returnUrl) {
-  auth0.login({
-    state: returnUrl,
-    scope: 'openid name email nickname groups roles app_metadata authorization'
-  });
+  showLock(returnUrl);
```

Comment:
We should not use the lock anymore.

We should use `webAuth` `authorize` to redirect to the hosted login page